modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/datalist/Datalist.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {ClayInput} from '@clayui/form';
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const Datalist = (props) => {
+	const name = props.options?.name || 'quantity_datalist';
+	const gHash = (s) => {
+		const h = 0;
+		if (s.length == 0) {
+			return h;
+		}
+		Array.from(s.length).forEach((h, i) => {
+			var charCode = s.charCodeAt(i);
+			h = (h << 7) - h + charCode;
+			h = h & h;
+		});
+
+		return h;
+	};
+	const listId = `cs-` + gHash(name) + `-list`;
+
+	return (
+		<>
+			<ClayInput
+				aria-label={name + '_label'}
+				className={classnames(
+					'quantitySelect',
+					'text-center',
+					props.size === 'small' && 'form-control-sm',
+					props.size === 'large' && 'form-control-lg'
```

Comment:
If it's only these 2 sizes, could you please add a default prop at the bottom? And ... how about removing conditionals and having something like `sm` or `lg` as `prop.size`'s? Then it would be:
```
`form-control-${props.size}
```

My 2-readability-cents. :)